power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -33,13 +42,63 @@ inline auto build_transformer_graph(State const& /*state*/) -> TransformerGraph
     return {};
 }
 
-inline auto get_edge_weights(TransformerGraph const& /*graph*/) -> std::vector<std::pair<Idx2D, EdgeWeight>> {
-    // TODO(mgovers): implement
-    return {};
+inline auto get_edge_weights(TransformerGraph const& graph) -> std::vector<std::pair<Idx2D, EdgeWeight>> {
+    // Step 2: Initialize the rank of all vertices (transformer nodes) as infinite (INT_MAX)
+    std::vector<EdgeWeight> rank(boost::num_vertices(graph), infty);
+    std::vector<Idx2D> sources(boost::num_vertices(graph));
+
+    // Step 3: Loop all the connected sources (status == 1)
+    for (auto v : boost::make_iterator_range(boost::vertices(graph))) {
```

Comment:
wouldn't it be possible to do the entire graph in 1 go? right now, complexity is `V(E + V log V)` (cfr. wikipedia; performing a full Dijkstra for each vertex) but I feel like you can keep track of the current best path on the fly. At some point, for all vertices, their neighbours have `weight_neighbour >= weight_vertex` and you are done for all vertices

Reply:
In `topology.hpp`, we use DFS (https://www.boost.org/doc/libs/1_84_0/libs/graph/doc/depth_first_search.html). My guess is that we can use BFS (https://cs.brown.edu/people/jwicks/boost/libs/graph/doc/breadth_first_search.html) to do this for us

Reply:
Will revisit. Indeed room for optimization.